src/Operation/ListCollections.php

```diff
@@ -68,10 +68,9 @@ public function __construct(private string $databaseName, array $options = [])
      * Execute the operation.
      *
      * @see Executable::execute()
-     * @return CollectionInfoIterator
      * @throws DriverRuntimeException for other driver errors (e.g. connection errors)
      */
-    public function execute(Server $server)
+    public function execute(Server $server): CollectionInfoIterator
```

Comment:
I started a [Slack thread](https://mongodb.slack.com/archives/C026UA4GF0B/p1726247537366849) to discuss whether these iterators and the corresponding model classes provide any value. Even if we keep the model classes, I suppose we could remove the iterators and rely on Psalm to type them as an iterator of a particular model class.